karma.py

```diff
@@ -0,0 +1,84 @@
+from errbot import BotPlugin, botcmd, re_botcmd
+
+
+KARMA_INC_REGEX = '^([a-zA-Z0-9_]+)((:?\+)+)$'
+KARMA_DEC_REGEX = '^([a-zA-Z0-9_]+)((:?\-)+)$'
+
+
+class Karma(BotPlugin):
+    def activate(self):
+        super(Karma, self).activate()
+        try:
+            if type(self['karma']) is not dict:
+                self['karma'] = {}
+        except KeyError:
+            self['karma'] = {}
+
+    def update_karma(self, msg, match, increment):
+        name = match.group(1)
+        update = match.group(2)
+        nick = msg.frm.nick
+        karmas = self['karma']
+
+        if name == nick:
+            return 'Not in this universe, maggot!'
+        else:
+            if name not in karmas:
+                karmas[name] = 0
+
+            if increment:
```

Comment:
@Adman I could see how we could improve this a bit by doing something like
```python
delta = len(update)
sign = -1 if increment else 1
karmas[name] += sign*delta
```

but it's probably not necessary as this is quite readable.